server/src/main/java/org/apache/druid/client/ImmutableDruidDataSource.java

```diff
@@ -120,8 +120,39 @@ public String toString()
            + "'}";
   }
 
+  /**
+   *
+   * ImmutableDruidDataSource should be considered a container, not a data class.
+   * The idea is the same as behind prohibiting/limiting equals() (and therefore usage as HashSet/HashMap keys) of DataSegment: see #6358.
```

Comment:
Use `https://github.com/apache/incubator-druid/issues/6358` instead of `#6358`. People cannot follow `#6358` when they see this comment in IDE.

Reply:
Fixed